dlib/media/ffmpeg_details.h

```diff
@@ -480,6 +480,39 @@ namespace dlib { namespace ffmpeg { namespace details
         }
     }    
 
+// ---------------------------------------------------------------------------------------------------
+
+    inline AVCodecID pick_codec_from_filename(const std::string& filename)
+    {
+        const auto ext_pos = filename.find_last_of(".");
+
+        if (ext_pos != std::string::npos)
+        {
+            const std::string ext = filename.substr(ext_pos + 1);
+
+            if (ext == "png" || ext == "PNG")
+                return AV_CODEC_ID_PNG;
+            else if (ext == "jpeg" || ext == "jpg" || ext == "JPEG")
+                return AV_CODEC_ID_MJPEG;
+            else if (ext == "tiff")
+                return AV_CODEC_ID_TIFF;
+            else if (ext == "webp")
+                return AV_CODEC_ID_WEBP;
+            else if (ext == "bmp")
+                return AV_CODEC_ID_BMP;
+            else if (ext == "h264")
+                return AV_CODEC_ID_H264;
+            else if (ext == "h265" || ext == "hevc")
+                return AV_CODEC_ID_H265;
+            else if (ext == "aac")
+                return AV_CODEC_ID_AAC;
+            else if (ext == "ac3")
+                return AV_CODEC_ID_AC3;
+        }
```

Comment:
How about adding JPEG XL support?
```suggestion
            else if (ext == "jxl")
                return AV_CODEC_ID_JPEGXL;
        }
```

Reply:
At some point, I would like to add proper JPEG XL support
- `rgb_pixel` / `rgba_pixel` images
- animations
- maybe other pixel formats (more pixel formats: bit depth and channels)

When I was working on WebP support, I already saw it was a nice improvement over JPEG, PNG, and GIF (I have a branch with partial support for WebP animations in dlib)

However, in practice, I often hit the limitations in size (it is a VP8 intra frame, so the width and height is stored in 14 bits, which means 16383 x 16383 pixels, at most).

Then I discovered JPEG XL and it is an outstanding image format. I mean, it's a proper image format, not a video frame repurposed as an image format, like WebP or AVIF.

I wish [Google added support back to Chrome](https://bugs.chromium.org/p/chromium/issues/detail?id=1178058) (not because I use Chrome, but because it will kill the format).

Sorry for the small rant...

Reply:
Only thing is that JPEGXL support didn't land until FFmpeg 5.1

Reply:
Thank you for adding that :)